File: daraja/src/main/java/com/androidstudy/daraja/callback/DarajaPaymentListener.java
package com.androidstudy.daraja.callback;

import com.androidstudy.daraja.model.PaymentResult;

public interface DarajaPaymentListener {

    void onPaymentRequestComplete(PaymentResult result);

    void onPaymentFailure(DarajaException exception);

    void onNetworkFailure(DarajaException exception);
}

File: daraja/src/main/java/com/androidstudy/daraja/callback/DarajaPaymentListener.kt
package com.androidstudy.daraja.callback

import com.androidstudy.daraja.model.PaymentResult

interface DarajaPaymentListener {
    fun onPaymentRequestComplete(result: PaymentResult)
    fun onPaymentFailure(exception: DarajaException)
    fun onNetworkFailure(exception: DarajaException)
}